File: cash.py
import cs50
from cs50 import get_float
from math import floor

def main():

#Inicio
    while True:
        troco_devido = get_float("Troco Devido: ")
        if troco_devido >= 0:
            break
    centavos = int(troco_devido * 100)
    print(cash(centavos))


#Calculo
def cash(centavos):
    return centavos // 25 + (centavos %25) // 10 + (centavos %25 %10) // 5 + (centavos %25 %10 %5) // 1 + centavos %25 %10 %5 %1
    #Resultado
if __name__ == "__main__":
     main()

File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    printf("\nSaudações - Dinheiro Trocado.\n"); 

    float din;
    do
    {
        din = get_float("\nDigite o valor do troco?: R$");
        
     }while (din < 0);
     
     int cents = round(din * 100);
     int moedas = 0;
     
     while (cents >= 25)
     {
         cents -= 25;
         moedas++;
     }
     while (cents >=10)
     {
         cents -= 10;
         moedas++;
     }
     while (cents >= 5)
     {
         cents -= 5;
         moedas++;
     }
     while (cents >= 1)
     {
         cents -= 1;
         moedas++;
     }
     
         printf("%i\n", moedas);
     
}